src/query/functions/temporal/base.go

```diff
@@ -256,15 +267,16 @@ func (c *baseNode) processCompletedBlocks(queryCtx *models.QueryContext, process
 	sp, _ := opentracingutil.StartSpanFromContext(queryCtx.Ctx, c.op.OpType())
 	defer sp.Finish()
 
-	blocks := make([]block.Block, len(processRequests))
-	for i, req := range processRequests {
+	blocks := make([]block.Block, 0, len(processRequests))
```

Comment:
Ah yeah this is the reason for the panic. Previously, I was allocating blocks with `length = len(processRequests)`, which meant that the array was:

```
[]block.Block{nil, nil, ...}
```

In case of early return (on error), we'd loop through every block and try to close it, causing a segfault.

Reply:
Ah gotcha; would it be worth adding nil checks into `closeBlocks`?

Reply:
Eh I don't think so; there's no valid use case for having a nil element in the slice. That probably means I should a check in the test though--good call!